### PR TITLE
Update xunit to 2.9.2

### PR DESCRIPTION
**Pull request by akoeplinger**
## Description

Updates the xunit version following https://github.com/dotnet/arcade/pull/15170

## PR Checklist

If you are upgrading the version of a repo submodule, please follow this
checklist:

- [ ] Provide a link to an issue in the consuming repo describing the need for
the upgrade. Both this PR and the PR doing the upgrade in the consuming repo
should link to that issue.
- [ ] Have you done your due diligence to ensure the upgrade can be completed
in the consuming repo in a timely manner? If consuming the dependency flow of
this update takes a long time or needs to be backed out, it may require the
reversion of the upgrade in this PR. That's something we want to avoid.
- [ ] When consuming the dependency flow from this repo for the purposes of a
version upgrade, consider using a separate PR or at least changing the title of
the dependency flow PR to accurately reflect the purpose of the change. Seeing
a PR named "Upgrade IdentityModel to 8.0.1", for example, provides better
clarity than "Update dependencies from dotnet/source-build-externals".

### CODE REVIEW

File: patches/xunit/0001-Patching.patch [hunk 1]
```diff
@@ -79,23 +46,37 @@ index 87c4ea80..7fdf2a3d 100644
...
+ 
+   <PropertyGroup>
+-    <DefineConstants Condition=" '$(TargetFramework)' == 'netstandard1.1' ">$(DefineConstants);XUNIT_NULLABLE</DefineConstants>
++    <DefineConstants Condition=" '$(TargetFramework)' == 'netstandard2.0' ">$(DefineConstants);XUNIT_NULLABLE</DefineConstants>
+     <DefineConstants Condition=" '$(TargetFramework)' == 'net6.0' ">$(DefineConstants);XUNIT_IMMUTABLE_COLLECTIONS;XUNIT_NULLABLE;XUNIT_SPAN</DefineConstants>
      <GenerateDocumentationFile>true</GenerateDocumentationFile>
      <Nullable>enable</Nullable>
--    <TargetFramework>netstandard1.1</TargetFramework>
+-    <TargetFrameworks>netstandard1.1;net6.0</TargetFrameworks>
 +    <TargetFramework>netstandard2.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (akoeplinger) [reply patches/xunit/0001-Patching.patch:80]: ah no that was just a mistake when updating the patch, I can revert it.

[the remark this answers is not in the document]
Comment (akoeplinger) [reply patches/xunit/0001-Patching.patch]: Fixed

**Review by mthalman — APPROVED** [review]

File: patches/xunit/0001-Patching.patch [hunk 2]
```diff
@@ -79,24 +46,38 @@ index 87c4ea80..7fdf2a3d 100644
...
+   <PropertyGroup>
+-    <DefineConstants Condition=" '$(TargetFramework)' == 'netstandard1.1' ">$(DefineConstants);XUNIT_NULLABLE</DefineConstants>
++    <DefineConstants Condition=" '$(TargetFramework)' == 'netstandard2.0' ">$(DefineConstants);XUNIT_NULLABLE</DefineConstants>
+     <DefineConstants Condition=" '$(TargetFramework)' == 'net6.0' ">$(DefineConstants);XUNIT_IMMUTABLE_COLLECTIONS;XUNIT_NULLABLE;XUNIT_SPAN</DefineConstants>
      <GenerateDocumentationFile>true</GenerateDocumentationFile>
      <Nullable>enable</Nullable>
--    <TargetFramework>netstandard1.1</TargetFramework>
-+    <TargetFramework>netstandard2.0</TargetFramework>
+-    <TargetFrameworks>netstandard1.1;net6.0</TargetFrameworks>
++    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
```
Comment (ViktorHofer) [inline patches/xunit/0001-Patching.patch:80]: Just to double check, @dotnet/source-build-internal is targeting net6.0 fine here? I see that there's a targeting pack for it in SBRP.

Comment (MichaelSimons) [reply patches/xunit/0001-Patching.patch:80]: Yes that is fine.

**Pull request merged**